DAOS-17259 test: DAOS build with gcov test.

dinghwah opened the pull request:
Test only.

### Steps for the author:

* [ ] Commit message follows the [guidelines](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Appropriate [Features or Test-tag](https://daosio.atlassian.net/wiki/spaces/DC/pages/10984259629/Test+Tags) pragmas were used.
* [ ] Appropriate [Functional Test Stages](https://daosio.atlassian.net/wiki/spaces/DC/pages/12147556353/CI+Functional+Test+Stages) were run.
* [ ] At least two positive code reviews including at least one code owner from each category referenced in the PR.
* [ ] Testing is complete. If necessary, forced-landing label added and a reason added in a comment.

#### After all prior steps are complete:
* [ ] Gatekeeper requested (daos-gatekeeper added as a reviewer).

dinghwah committed:
DAOS-17259 test: DAOS build with gcov test.

Test only.

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild1 commented:
Test stage Build RPM on Leap 15.5 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/1/execution/node/283/log

daosbuild1 commented:
Test stage Build on Leap 15.5 with Intel-C and TARGET_PREFIX completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/1/execution/node/398/log

daosbuild1 commented:
Test stage Build on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/1/execution/node/403/log

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/1/execution/node/369/log

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/1/execution/node/261/log

daosbuild1 commented:
Test stage Build RPM on EL 9 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/1/execution/node/372/log

dinghwah committed:
Test only, update SConscript with test_coverage.

Skip-test: true
Skip-unit-tests: true
Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild1 commented:
Test stage Build RPM on Leap 15.5 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/2/execution/node/343/log

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/2/execution/node/344/log

daosbuild1 commented:
Test stage Build RPM on EL 9 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/2/execution/node/345/log

daosbuild1 commented:
Test stage Build on Leap 15.5 with Intel-C and TARGET_PREFIX completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/2/execution/node/507/log

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest

daosbuild1 commented:
Test stage Build RPM on Leap 15.5 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/3/execution/node/310/log

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/3/execution/node/303/log

daosbuild1 commented:
Test stage Build on Leap 15.5 with Intel-C and TARGET_PREFIX completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/3/execution/node/409/log

daosbuild1 commented:
Test stage Build on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/3/execution/node/448/log

dinghwah committed:
Test only, update SConscript for gcov build

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild1 commented:
Test stage Build on Leap 15.5 with Intel-C and TARGET_PREFIX completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/4/execution/node/315/log

daosbuild1 commented:
Test stage Build RPM on EL 9 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/4/execution/node/300/log

daosbuild1 commented:
Test stage Build RPM on Leap 15.5 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/4/execution/node/299/log

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/4/execution/node/354/log

dinghwah committed:
Test only, update SConscript

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild1 commented:
Test stage Build on Leap 15.5 with Intel-C and TARGET_PREFIX completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/5/execution/node/337/log

daosbuild1 commented:
Test stage Build RPM on EL 9 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/5/execution/node/322/log

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/5/execution/node/325/log

daosbuild1 commented:
Test stage Build RPM on Leap 15.5 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/5/execution/node/316/log

dinghwah committed:
Test only, Update SConscript

Skip-build-Leap15: true
Test-Tag: test_pool_connect
Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild1 commented:
Test stage Build on Leap 15.5 with Intel-C and TARGET_PREFIX completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/6/execution/node/497/log

dinghwah committed:
Test only, update src/common/dav_v2/SConscript

Test-tag: test_pool_connect
Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild1 commented:
Test stage Build on Leap 15.5 with Intel-C and TARGET_PREFIX completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/7/execution/node/343/log

daosbuild1 commented:
Test stage Build RPM on EL 9 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/7/execution/node/331/log

dinghwah committed:
Test only, recommit to skip Leap15.5

Skip-build-Leap15-rpm: true
Skip-build-Leap15: true
Test-tag: test_pool_connect
Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/7/execution/node/323/log

daosbuild1 commented:
Test stage Build on Leap 15.5 with Intel-C and TARGET_PREFIX completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/8/execution/node/322/log

daosbuild1 commented:
Test stage Build RPM on EL 9 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/8/execution/node/300/log

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/8/execution/node/294/log

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest

dinghwah committed:
Update src/common/dav_v2/SConscript.

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update src/common/tests/SConscript

Skip-build-Leap15-icc: true
Skip-build-Leap15-gcc: true
Skip-build-Leap15-rpm: true
Test-tag: test_pool_connect
Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest

daosbuild1 commented:
Test stage Build RPM on EL 9 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/9/execution/node/313/log

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/9/execution/node/307/log

daosbuild1 commented:
Test stage Build RPM on Leap 15.5 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/9/execution/node/310/log

daosbuild1 commented:
Test stage Build on Leap 15.5 with Intel-C and TARGET_PREFIX completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/9/execution/node/448/log

dinghwah committed:
Update src/common/dav_v2/SConscript

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild1 commented:
Test stage Build on Leap 15.5 with Intel-C and TARGET_PREFIX completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/10/execution/node/339/log

daosbuild1 commented:
Test stage Build RPM on EL 9 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/10/execution/node/330/log

daosbuild1 commented:
Test stage Build RPM on Leap 15.5 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/10/execution/node/327/log

dinghwah committed:
Set --test-coverage default False (disable) Test only.

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Restore SConscript with default True for --test-coverage.

Skip-build-leap15-icc-debug: true
Test-tag: test_pool_connect
Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild1 commented:
Test stage Build on Leap 15.5 with Intel-C and TARGET_PREFIX completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/12/execution/node/329/log

daosbuild1 commented:
Test stage Build RPM on EL 9 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/12/execution/node/323/log

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/12/execution/node/304/log

daosbuild1 commented:
Test stage Build RPM on Leap 15.5 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/12/execution/node/322/log

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest

dinghwah committed:
Update and recommit to skip leap15.

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Test-tag: test_pool_connect

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update site_con/compiler_setup.py with -g -O0

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Test-tag: test_pool_connect

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest

dinghwah committed:
Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Test-tag: test_pool_connect

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update to generate code_coverage_report.html

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Skip-nlt: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update code_coverage_report path.

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Skip-nlt: true
Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update gcovr to ci/unit/required_packages.sh

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Skip-nlt: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild1 commented:
Test stage Unit Test bdev with memcheck on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-16118/17/display/redirect

daosbuild1 commented:
Test stage Unit Test on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-16118/17/display/redirect

daosbuild1 commented:
Test stage Unit Test bdev on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-16118/17/display/redirect

dinghwah committed:
Update gcovr for pip install

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Skip-nlt: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
update requirements-build.txt with gcovr

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Skip-nlt: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Relaunch test

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Skip-nlt: true
Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
retest

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Skip-nlt: true
Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild1 commented:
Test stage Unit Test bdev with memcheck on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-16118/21/display/redirect

daosbuild1 commented:
Test stage Unit Test on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-16118/21/display/redirect

daosbuild1 commented:
Test stage Unit Test bdev on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-16118/21/display/redirect

daosbuild1 commented:
Test stage Unit Test with memcheck on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-16118/21/display/redirect

dinghwah committed:
Test.

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Skip-nlt: true
Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update to take care of Pylint error.

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Skip-nlt: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update /usr/bin/gcovr

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Skip-nlt: true
Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update with temp pip3 install gcovr.

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Skip-nlt: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

phender commented on requirements-utest.txt:
```diff
@@ -6,3 +6,4 @@ PyYAML
 # pytest
 # for pydaos.torch module
 torch
+gcovr
```
This should be specified in ci/unit/required_packages.sh
```suggestion
```

phender commented on requirements-ftest.txt:
```diff
@@ -5,3 +5,4 @@ clustershell
 paramiko
 distro
 torch
+gcovr
```
We don't need this at the moment.
```suggestion
```

phender commented on ci/unit/test_post_always_node.sh:
```diff
@@ -18,3 +18,11 @@ if ! sudo umount "$DAOS_BASE"; then
+
+# Run code coverage gcovr
+if find build -name "./*.gcda"; then
+  # tmp pip3 install
+  sudo pip3 install gcovr
```
This should be handled by an entry in ci/unit/required_packages.sh
```suggestion
```

dinghwah committed:
Update gcovr report to test_main_node.sh

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Skip-nlt: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update gcovr to test_main_node.sh

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Skip-nlt: true
Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update gcovr to test_main_node.sh

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Skip-nlt: true
Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild1 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-16118/27/testReport/

dinghwah committed:
Update test_main_node.sh

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Skip-nlt: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild1 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-16118/28/testReport/

daosbuild1 commented:
Test stage Unit Test with memcheck on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/28/execution/node/585/log

dinghwah committed:
Update ci/unit/test_main_node.sh

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Skip-nlt: true
Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update ci/unit/test_main_node.sh

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Skip-nlt: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild1 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-16118/30/testReport/

dinghwah committed:
Update origin/dinghwah/DAOS-17259-gtest

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Skip-nlt: true
Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild1 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-16118/31/testReport/

dinghwah committed:
Update ci/unit/test_main_node.sh

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Skip-nlt: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild1 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-16118/32/testReport/

dinghwah committed:
Update ci/unit/test_main_node.sh

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Skip-nlt: true
Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild1 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-16118/33/testReport/

dinghwah committed:
Update ci/unit/test_main_node.sh

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Skip-nlt: true
Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild1 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-16118/34/testReport/

daosbuild1 commented:
Test stage Unit Test bdev on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/34/execution/node/583/log

daosbuild1 commented:
Test stage Unit Test with memcheck on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/34/execution/node/556/log

daosbuild1 commented:
Test stage Unit Test bdev with memcheck on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/34/execution/node/584/log

dinghwah committed:
Update ci/unit/test_main_node.sh

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Skip-nlt: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild1 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-16118/35/testReport/

daosbuild1 commented:
Test stage Unit Test bdev on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/35/execution/node/556/log

daosbuild1 commented:
Test stage Unit Test with memcheck on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/35/execution/node/584/log

daosbuild1 commented:
Test stage Unit Test bdev with memcheck on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/35/execution/node/555/log

dinghwah committed:
Update ci/unit/test_main_node.sh with debug

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Skip-nlt: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update ci/unit/test_main_node.sh with debug

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Skip-nlt: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild1 commented:
Test stage Unit Test bdev on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/37/execution/node/583/log

daosbuild1 commented:
Test stage Unit Test with memcheck on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/37/execution/node/584/log

dinghwah committed:
debug before anf after run-utest

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-func-test: true
Skip-nlt: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Add debug

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest

daosbuild1 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-16118/39/testReport/

daosbuild1 commented:
Test stage Unit Test bdev on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/39/execution/node/391/log

dinghwah committed:
Update debug

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Debug gcno after build

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild1 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-16118/43/testReport/

daosbuild1 commented:
Test stage Unit Test bdev on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/43/execution/node/391/log

daosbuild1 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-16118/44/testReport/

daosbuild1 commented:
Test stage Unit Test bdev on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/44/execution/node/401/log

dinghwah committed:
Restore test_file_to_stash

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

dinghwah committed:
Update with html-details

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild1 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-16118/45/testReport/

daosbuild1 commented:
Test stage Unit Test bdev on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/45/execution/node/399/log

dinghwah committed:
Add code coverage json file

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild1 commented:
Test stage Unit Test on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/46/execution/node/384/log

daosbuild1 commented:
Test stage Unit Test bdev on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/46/execution/node/401/log

dinghwah committed:
Update gcovr with html w/o detail

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

phender committed:
Generate a single code coverage reort.

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-func-test: true

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

daosbuild1 commented:
Test stage Unit Test on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/47/execution/node/401/log

phender committed:
Add code coverage report generation.

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-func-test: true

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

daosbuild1 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-16118/49/testReport/

phender committed:
Don't rsync venev files.  Add debug.

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-func-test: true

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest

phender committed:
Build with python 3.11

Skip-build-el9: false
Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender commented on ci/unit/test_post_always.sh:
```diff
@@ -39,11 +39,11 @@ esac
 # standard wildcards.
 rsync -v -dpt -z -e "ssh $SSH_KEY_ARGS" jenkins@"$NODE":/tmp/ \
-      --filter="include daos*.log" --filter="include test.cov" \
+      --filter="include daos*.log" --filter="include code_coverage/*" \
```
Eventually remove this once a single code coverage file working.

phender committed:
Remove building an running UT w/ python 3.11

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender commented on ci/unit/test_main_node.sh:
```diff
@@ -77,14 +77,23 @@ rm -rf "$test_log_dir"
+# Generate code coverage report
+if [[ -n $(find build -name "*.gcda") ]]; then
+    mkdir -p "${test_log_dir}/code_coverage"
+    gcovr -o "${test_log_dir}/code_coverage/code_coverage_report.html" --html-details --gcov-ignore-parse-errors
```
Eventually remove this one and only generate json files per stage.

phender committed:
Add Code CCoverage Report stage

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Fix Jenkinsfile

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Fix.

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Jenkinsfile update.

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

dinghwah committed:
Update SConscript copy right to recommit

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: false
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

phender committed:
Attempt to get push to work.

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Update docker related files.

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest

phender committed:
Always run gcovr with the same python version.

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

daosbuild1 commented:
Test stage Unit Test on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/58/execution/node/392/log

dinghwah committed:
Run unit-tests

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-nlt: false
Skip-unit-test-memcheck: flase
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild1 commented:
Test stage Unit Test on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/59/execution/node/481/log

daosbuild1 commented:
Test stage Unit Test bdev on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/59/execution/node/419/log

daosbuild1 commented:
Test stage Unit Test with memcheck on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/59/execution/node/415/log

phender committed:
Fix new venv name.

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

daosbuild1 commented:
Test stage Unit Test on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/60/execution/node/392/log

daosbuild1 commented:
Test stage Unit Test bdev on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/60/execution/node/378/log

phender committed:
Avoid unbound variable failure.

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender committed:
Remove find -quit arg.

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

dinghwah committed:
Comment out ls al on code coverage report

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-nlt: false
Skip-unit-test-memcheck: false
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

phender committed:
Merge branch 'master' into dinghwah/DAOS-17259-gtest

phender committed:
Add support for NLT

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-nlt: false
Skip-unit-test-memcheck: false
Skip-func-test: true

Signed-off-by: Phil Henderson <phillip.henderson@hpe.com>

phender commented on Jenkinsfile:
```diff
@@ -813,7 +813,8 @@ pipeline {
                             unitTestPost artifacts: ['unit_test_memcheck_logs.tar.gz',
-                                                     'unit_test_memcheck_logs/**/*.log'],
+                                                     'unit_test_memcheck_logs/**/*.log',
+                                                     'unit_test_memcheck_logs/code_coverage'],
```
For debug only

phender commented on Jenkinsfile:
```diff
@@ -838,7 +839,8 @@ pipeline {
                             unitTestPost artifacts: ['unit_test_memcheck_bdev_logs.tar.gz',
-                                                     'unit_test_memcheck_bdev_logs/**/*.log'],
+                                                     'unit_test_memcheck_bdev_logs/**/*.log',
+                                                     'unit_test_memcheck_logs/code_coverage'],
```
For debug only

phender commented on ci/unit/test_nlt_node.sh:
```diff
@@ -43,3 +43,12 @@ prlimit -n
+    pip install --requirement requirements-code-coverage.txt
+
+    mkdir -p nlt_logs/code_coverage
+    gcovr -o nlt_logs/code_coverage/code_coverage_report.html --html-details --gcov-ignore-parse-errors
```
Debug

dinghwah committed:
Add debug for unit-test NLT

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-unit-test: true
Skip-nlt: false
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

jolivier23 commented on src/client/dfuse/SConscript:
```diff
@@ -101,7 +105,10 @@ def build_client_libs_shared(env, prereqs):
 
-    gen_script = ilenv.d_program('il/gen_script', ['il/gen_script.c'], LIBS=[])
+    libs = []
+    if GetOption("test_coverage"):
```
Just a thought.  Rather than sprinkling these checks everywhere, why not just do something in the various builders

https://github.com/daos-stack/daos/blob/367d2667a213b50ddda8ae479580f417bc5adf96/site_scons/site_tools/daos_builder.py#L167

We call d_program, d_test_program, d_library, etc all over the place.  If you just put an Append inside those routines, you can limit this to a few LOC

Those functions are added here: https://github.com/daos-stack/daos/blob/367d2667a213b50ddda8ae479580f417bc5adf96/site_scons/site_tools/daos_builder.py#L251-L261

dinghwah committed:
Update per Jeff's comment.

Skip-build-leap15-icc: true
Skip-build-leap15-gcc: true
Skip-build-leap15-rpm: true
Skip-build-el8-rpm: true
Skip-build-el9-rpm: true
Skip-unit-test: false
Skip-nlt: true
Skip-unit-test-memcheck: true
Skip-func-test: true

Signed-off-by: Ding-Hwa Ho <ding-hwa.ho@hpe.com>

daosbuild1 commented:
Test stage Build on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-16118/66/execution/node/250/log
